modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/rule/logical/ExposeIndexRule.java

```diff
@@ -51,8 +53,7 @@ public ExposeIndexRule(Config config) {
 
     /** */
     private static boolean preMatch(IgniteLogicalTableScan scan) {
-        return scan.simple() // was not modified by ProjectScanMergeRule or FilterScanMergeRule
```

Comment:
It was the only usage of the `simple()` method, perhaps this method should be removed too.

Reply:
fixed